### PR TITLE
Use phpstan-drupal for code analysis

### PR DESCRIPTION
**Pull request by jekuaitk**
- Add phpstan-drupal without drupal-check package
- Fix coding errors

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -81,7 +81,10 @@
     "require-dev": {
         "drupal/core-dev": "^9.1",
         "drupal/mail_debugger": "^1.4",
-        "mglaman/drupal-check": "^1.3"
+        "mglaman/phpstan-drupal": "^1.1",
+        "phpstan/extension-installer": "^1.2",
```
Comment (rimi-itk) [inline composer.json:85]: Didn't we agree to not use this and, hence, to manually and explicitly enable PHPStan extensions?

File: composer.json [hunk 2]
```diff
@@ -190,11 +194,8 @@
         "install-codestandards": [
             "Dealerdirect\\Composer\\Plugin\\Installers\\PHPCodeSniffer\\Plugin::run"
         ],
-        "code-analysis/drupal-check": [
-            "vendor/bin/drupal-check --deprecations --analysis web/modules/custom"
-        ],
         "code-analysis": [
-            "@code-analysis/drupal-check"
```
Comment (rimi-itk) [inline composer.json:197]: I think we rename to `"code-analysis/phpstan"` and keep `"code-analysis"` (almost) as it was (with an updated name), i.e.

```json
        "code-analysis/phpstan": [
            "vendor/bin/phpstan analyse"
        ],
        "code-analysis": [
            "@code-analysis/phpstan"
        ],
```

This is how we do it in all/most projects.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (martinyde) [reply composer.json:85]: I'm unsure what we agreed upon. I might have misunderstood. As long as we have a phpstan.neon config file to set it all up i think extension-installer is ok, It cleans up the composer file a bit but is less explicite which might be a bad thing.

File: composer.json (hunk 2 above)
Comment (martinyde) [reply composer.json]: Fixed

File: composer.json (hunk 1 above)
Comment (rimi-itk) [reply composer.json:85]: As far as I recall we agreed on being explicit, but if phpstan/extension-installer works as intended (and does the right things) I'm fine with using it. https://github.com/phpstan/extension-installer#motivation argues that it is indeed helpful.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**